lib/ansible/modules/cloud/openstack/os_quota.py

```diff
@@ -309,10 +309,18 @@ def _get_compute_quotas(cloud, project):
 
 def _get_quotas(cloud, project):
 
+    get_quota_map = {
+        'volume':  _get_volume_quotas,
+        'network': _get_network_quotas,
+        'compute': _get_compute_quotas,
+    }
+
     quota = {}
-    quota['volume'] = _get_volume_quotas(cloud, project)
-    quota['network'] = _get_network_quotas(cloud, project)
-    quota['compute'] = _get_compute_quotas(cloud, project)
+    for q, fn in get_quota_map:
```

Comment:
@gandelman-a Is there a better way we could handle missing endpoints? I think it might be best if we could determine which endpoints should exist from an API call first and the loop over that. My concern is with the way the code is written below would we skip applying a code quota if a service was down or in a broken state? 

Maybe attempting to catch a specific exception instead of a blanket pass?

Reply:
One better option would be to determine which APIs are required based on the module parameters being passed in.  E.g., setting instances=5 would require and validate the nova API is there, setting volumes=5 would require cinder API is there, etc.

Reply:
The downside of that is you would need to implement some form of key/value to endpoint mapping which feels like a pain to manage. 

I believe that shade has a list_endpoints method (https://docs.openstack.org/shade/latest/user/usage.html), can we use that to determine which endpoints to query?